src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/MethodAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MethodAnalyzer
+    {
+        public const string DiagnosticId = "DF0107";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(Resources.MethodAnalyzerTitle), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(Resources.MethodAnalyzerMessageFormat), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(Resources.DeterministicAnalyzerDescription), Resources.ResourceManager, typeof(Resources));
+        private const string Category = SupportedCategories.Orchestrator;
+        public const DiagnosticSeverity severity = DiagnosticSeverity.Warning;
+
+        public static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
+
+        public static bool RegisterDiagnostics(SyntaxNode methodDeclaration, CompilationAnalysisContext context, SemanticModel semanticModel)
+        {
+            var methodInformationList = AnalyzeMethod(methodDeclaration, context, semanticModel);
+            methodInformationList = RegisterAnalyzers(methodInformationList, context, semanticModel);
+            return DiagnoseInvocations(methodInformationList, context, semanticModel);
+        }
+
+        private static List<MethodInformation> RegisterAnalyzers(List<MethodInformation> methodInformationList, CompilationAnalysisContext context, SemanticModel semanticModel)
+        {
+            foreach (MethodInformation method in methodInformationList)
+            {
+                var declaration = method.Declaration;
+
+                if (DateTimeAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
+                    EnvironmentVariableAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
+                    GuidAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
+                    IOTypesAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
+                    ThreadTaskAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
+                    TimerAnalyzer.RegisterDiagnostic(declaration, context, semanticModel) |
```

Comment:
Are you intentionally use `|` instead of `||` here?  It seems like there could be a negative performance impact when using `|` instead of `||` as your logical OR operator if you don't actually need to.

Reply:
Yes, I intentionally use | instead of ||, and included a test for it. Here, I want all of the "analyzers" to run the code, the bool tells me if at least one produced a diagnostic.

Reply:
Cool. I suggest adding a comment explaining that this is intentional. I've never actually seen anyone do this before in C# so it might not be obvious to future readers of this code (for example, I could see someone changing it not knowing it was done intentionally).